In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/inc-5000-companies/INC 5000 Companies 2019.csv


In [2]:
#importing Dataset from kaggle in CSV format
data=pd.read_csv("/kaggle/input/inc-5000-companies/INC 5000 Companies 2019.csv")

In [3]:
companies=data.copy()
companies.head()

rank                                   profile               name  \
0     1      https://www.inc.com/profile/freestar           Freestar   
1     2   https://www.inc.com/profile/freightwise        FreightWise   
2     3  https://www.inc.com/profile/ceces-veggie  Cece's Veggie Co.   
3     4      https://www.inc.com/profile/ladyboss           LadyBoss   
4     5        https://www.inc.com/profile/perpay             Perpay   

                         url state       revenue    growth_%  \
0        http://freestar.com    AZ  36.9 Million  36680.3882   
1  http://freightwisellc.com    TN  33.6 Million  30547.9317   
2   http://cecesveggieco.com    TX  24.9 Million  23880.4852   
3        http://ladyboss.com    NM  32.4 Million  21849.8925   
4          http://perpay.com    PA  22.5 Million  18166.4070   

                       industry  workers  previous_workers  founded  \
0       Advertising & Marketing     40.0                 5     2015   
1    Logistics & Transportation     39.0                 8     2015   
2               Food & Beverage    190.0                10     2015   
3  Consumer Products & Services     57.0                 2     2014   
4                        Retail     25.0                 6     2014   

   yrs_on_list         metro          city  
0            1       Phoenix       Phoenix  
1            1     Nashville     Brentwood  
2            1        Austin        Austin  
3            1           NaN   Albuquerque  
4            1  Philadelphia  Philadelphia

In [4]:
#checking our data
companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5012 entries, 0 to 5011
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   rank              5012 non-null   int64  
 1   profile           5012 non-null   object 
 2   name              5012 non-null   object 
 3   url               5012 non-null   object 
 4   state             5012 non-null   object 
 5   revenue           5012 non-null   object 
 6   growth_%          5012 non-null   float64
 7   industry          5012 non-null   object 
 8   workers           5011 non-null   float64
 9   previous_workers  5012 non-null   int64  
 10  founded           5012 non-null   int64  
 11  yrs_on_list       5012 non-null   int64  
 12  metro             4199 non-null   object 
 13  city              5012 non-null   object 
dtypes: float64(2), int64(4), object(8)
memory usage: 548.3+ KB


In [5]:
#checking the column names in the data
companies.columns

Index(['rank', 'profile', 'name', 'url', 'state', 'revenue', 'growth_%',
       'industry', 'workers', 'previous_workers', 'founded', 'yrs_on_list',
       'metro', 'city'],
      dtype='object')

In [6]:
#creating a dataset containing null values.
companies[companies["metro"].isnull()]

rank                                            profile  \
3        4               https://www.inc.com/profile/ladyboss   
20      21  https://www.inc.com/profile/tailor-made-compou...   
21      22                  https://www.inc.com/profile/sd3it   
25      26  https://www.inc.com/profile/pegasus-optimizati...   
28      29                https://www.inc.com/profile/lumenad   
...    ...                                                ...   
4991  4980             https://www.inc.com/profile/msnw-group   
4995  4984                  https://www.inc.com/profile/dymax   
5001  4990     https://www.inc.com/profile/smart-energy-today   
5009  4998          https://www.inc.com/profile/naval-systems   
5011  5000                https://www.inc.com/profile/vivayic   

                               name                               url state  \
3                          LadyBoss               http://ladyboss.com    NM   
20          Tailor Made Compounding  http://tailormadecompounding.com    KY   
21                            SD3IT                  http://sd3it.com    FL   
25    Pegasus Optimization Managers                     pegasus15.com    TX   
28                          LumenAd                       lumenad.com    MT   
...                             ...                               ...   ...   
4991                     MSNW Group                          msnw.org    WA   
4995                          Dymax                         dymax.com    CT   
5001             Smart Energy Today              smartenergytoday.net    WA   
5009                  Naval Systems                          n-s-i.us    MD   
5011                        Vivayic                       vivayic.com    NE   

            revenue    growth_%                      industry  workers  \
3      32.4 Million  21849.8925  Consumer Products & Services     57.0   
20     10.2 Million   8327.0956                        Health     45.0   
21     11.9 Million   8282.8668             Computer Hardware     12.0   
25    128.2 Million   7426.8072                        Energy    191.0   
28     11.8 Million   7045.3631                      Software     60.0   
...             ...         ...                           ...      ...   
4991   16.5 Million     52.7413  Business Products & Services    467.0   
4995    104 Million     52.6344                 Manufacturing    377.0   
5001   20.5 Million     52.5381                        Energy     65.0   
5009   29.7 Million     52.2037           Government Services    187.0   
5011    4.5 Million     52.1691  Business Products & Services     27.0   

      previous_workers  founded  yrs_on_list metro             city  
3                    2     2014            1   NaN      Albuquerque  
20                   1     2015            1   NaN    Nicholasville  
21                   2     2014            1   NaN     The Villages  
25                  14     2015            1   NaN  College Station  
28                   3     2014            1   NaN         Missoula  
...                ...      ...          ...   ...              ...  
4991               357     1995            9   NaN         Ferndale  
4995               274     1980            9   NaN       Torrington  
5001                92     2012            6   NaN         Tumwater  
5009               127     2004            1   NaN   LEXINGTON PARK  
5011                22     2006            4   NaN          Lincoln  

[813 rows x 14 columns]

 We have two types of values in revenue column
1.  revenue in Million
2.  revenue in Billion

To determine the average we must have values in either of one unit.

In [7]:
# We are splitting the revenue field in two different fields 
companies[["revenue","Unit"]] = companies["revenue"].str.split(" ", expand= True)

In [8]:
companies.head()

rank                                   profile               name  \
0     1      https://www.inc.com/profile/freestar           Freestar   
1     2   https://www.inc.com/profile/freightwise        FreightWise   
2     3  https://www.inc.com/profile/ceces-veggie  Cece's Veggie Co.   
3     4      https://www.inc.com/profile/ladyboss           LadyBoss   
4     5        https://www.inc.com/profile/perpay             Perpay   

                         url state revenue    growth_%  \
0        http://freestar.com    AZ    36.9  36680.3882   
1  http://freightwisellc.com    TN    33.6  30547.9317   
2   http://cecesveggieco.com    TX    24.9  23880.4852   
3        http://ladyboss.com    NM    32.4  21849.8925   
4          http://perpay.com    PA    22.5  18166.4070   

                       industry  workers  previous_workers  founded  \
0       Advertising & Marketing     40.0                 5     2015   
1    Logistics & Transportation     39.0                 8     2015   
2               Food & Beverage    190.0                10     2015   
3  Consumer Products & Services     57.0                 2     2014   
4                        Retail     25.0                 6     2014   

   yrs_on_list         metro          city     Unit  
0            1       Phoenix       Phoenix  Million  
1            1     Nashville     Brentwood  Million  
2            1        Austin        Austin  Million  
3            1           NaN   Albuquerque  Million  
4            1  Philadelphia  Philadelphia  Million

In [9]:
companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5012 entries, 0 to 5011
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   rank              5012 non-null   int64  
 1   profile           5012 non-null   object 
 2   name              5012 non-null   object 
 3   url               5012 non-null   object 
 4   state             5012 non-null   object 
 5   revenue           5012 non-null   object 
 6   growth_%          5012 non-null   float64
 7   industry          5012 non-null   object 
 8   workers           5011 non-null   float64
 9   previous_workers  5012 non-null   int64  
 10  founded           5012 non-null   int64  
 11  yrs_on_list       5012 non-null   int64  
 12  metro             4199 non-null   object 
 13  city              5012 non-null   object 
 14  Unit              5012 non-null   object 
dtypes: float64(2), int64(4), object(9)
memory usage: 587.5+ KB


Datatype of revenue field is object, to convert values representing billions in millions we need to perform some numerical operation this can only be done on numeric data

In [10]:
# Converting data type of field
companies["revenue"] = pd.to_numeric(companies["revenue"])

In [11]:
companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5012 entries, 0 to 5011
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   rank              5012 non-null   int64  
 1   profile           5012 non-null   object 
 2   name              5012 non-null   object 
 3   url               5012 non-null   object 
 4   state             5012 non-null   object 
 5   revenue           5012 non-null   float64
 6   growth_%          5012 non-null   float64
 7   industry          5012 non-null   object 
 8   workers           5011 non-null   float64
 9   previous_workers  5012 non-null   int64  
 10  founded           5012 non-null   int64  
 11  yrs_on_list       5012 non-null   int64  
 12  metro             4199 non-null   object 
 13  city              5012 non-null   object 
 14  Unit              5012 non-null   object 
dtypes: float64(3), int64(4), object(8)
memory usage: 587.5+ KB


In [12]:
#creating copy of the data and storing it in new variable.
avg_revenue = companies.copy()

To convert the values in billions into millions we must first determine at what index we are having entries of values in Billions.

In [13]:
# creating a list of the index at which the revenue represents Billion value
lst = avg_revenue[avg_revenue["Unit"] == "Billion"].index
lst         

Int64Index([ 605,  914, 1076, 1621, 2602, 2664, 2730, 2793, 3278, 3605, 3690,
            3781, 3856, 3924, 4010, 4348, 4377, 4620, 4804, 4812, 4938, 4979,
            4985],
           dtype='int64')

In [14]:
# using for loop, multiplying revenue value at respective index with 1000
for i in lst:
    avg_revenue["revenue"].iloc[i]  = avg_revenue["revenue"].iloc[i] * 1000

/opt/conda/lib/python3.7/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_block(indexer, value, name)


In [15]:
#checking whether, the operation performed is correct and as per the requirement.
avg_revenue[avg_revenue["Unit"] == "Billion"]

rank                                            profile  \
605    605                  https://www.inc.com/profile/chewy   
914    913      https://www.inc.com/profile/uber-technologies   
1076  1073     https://www.inc.com/profile/prime-therapeutics   
1621  1617       https://www.inc.com/profile/allied-universal   
2602  2594                   https://www.inc.com/profile/epiq   
2664  2656    https://www.inc.com/profile/fortis-construction   
2730  2722  https://www.inc.com/profile/kodiak-building-pa...   
2793  2785                 https://www.inc.com/profile/strike   
3278  3270                 https://www.inc.com/profile/victra   
3605  3596  https://www.inc.com/profile/transportation-ins...   
3690  3681  https://www.inc.com/profile/odyssey-logistics-...   
3781  3772            https://www.inc.com/profile/coadvantage   
3856  3847   https://www.inc.com/profile/carahsoft-technology   
3924  3915       https://www.inc.com/profile/srs-distribution   
4010  4001         https://www.inc.com/profile/nfi-industries   
4348  4337            https://www.inc.com/profile/ga-partners   
4377  4366      https://www.inc.com/profile/consumer-cellular   
4620  4609  https://www.inc.com/profile/total-quality-logi...   
4804  4793  https://www.inc.com/profile/big-d-construction...   
4812  4801             https://www.inc.com/profile/readerlink   
4938  4927      https://www.inc.com/profile/84-lumber-company   
4979  4968      https://www.inc.com/profile/atlas-oil-company   
4985  4974  https://www.inc.com/profile/pacific-dental-ser...   

                                name                               url state  \
605                            Chewy                         chewy.com    FL   
914                Uber Technologies             http://uber.com/ca/en    CA   
1076              Prime Therapeutics      http://primetherapeutics.com    MN   
1621                Allied Universal                    http://aus.com    CA   
2602                            Epiq                    epiqglobal.com    GA   
2664             Fortis Construction            fortisconstruction.com    OR   
2730        Kodiak Building Partners                      kodiakbp.com    CO   
2793                          Strike              http://strikeusa.com    TX   
3278                          Victra                 http://victra.com    NC   
3605          Transportation Insight  http://transportationinsight.com    NC   
3690  Odyssey Logistics & Technology              odysseylogistics.com    CT   
3781                     CoAdvantage                   coadvantage.com    FL   
3856            Carahsoft Technology              http://carahsoft.com    VA   
3924                SRS Distribution        http://srsdistribution.com    TX   
4010                  NFI Industries                 nfiindustries.com    NJ   
4348                    G&A Partners            http://gnapartners.com    TX   
4377               Consumer Cellular              consumercellular.com    OR   
4620         Total Quality Logistics                           tql.com    OH   
4804        Big-D Construction Corp.                         big-d.com    UT   
4812                      Readerlink                    readerlink.net    IL   
4938               84 Lumber Company                      84lumber.com    PA   
4979               Atlas Oil Company                      atlasoil.com    MI   
4985         Pacific Dental Services         pacificdentalservices.com    CA   

      revenue  growth_%                      industry   workers  \
605    3500.0  735.5436                        Retail    9833.0   
914   11300.0  464.9123  Consumer Products & Services   22263.0   
1076  21400.0  390.5654                        Health    3300.0   
1621   5800.0  255.1902                      Security  155000.0   
2602   1000.0  150.7917  Business Products & Services    5800.0   
2664   1200.0  146.4234                  Construction     330.0   
2730   1100.0  141.2369                  Construction  

In [16]:
# as all the values now in the revenue column is in millions, renaming this field as revenue in Millions
avg_revenue = avg_revenue.rename(columns={"revenue" : "revenue in Millions"})
avg_revenue

rank                                            profile  \
0        1               https://www.inc.com/profile/freestar   
1        2            https://www.inc.com/profile/freightwise   
2        3           https://www.inc.com/profile/ceces-veggie   
3        4               https://www.inc.com/profile/ladyboss   
4        5                 https://www.inc.com/profile/perpay   
...    ...                                                ...   
5007  4996   https://www.inc.com/profile/village-plumbing-air   
5008  4997  https://www.inc.com/profile/real-restoration-g...   
5009  4998          https://www.inc.com/profile/naval-systems   
5010  4999            https://www.inc.com/profile/hnm-systems   
5011  5000                https://www.inc.com/profile/vivayic   

                        name                        url state  \
0                   Freestar        http://freestar.com    AZ   
1                FreightWise  http://freightwisellc.com    TN   
2          Cece's Veggie Co.   http://cecesveggieco.com    TX   
3                   LadyBoss        http://ladyboss.com    NM   
4                     Perpay          http://perpay.com    PA   
...                      ...                        ...   ...   
5007  Village Plumbing & Air        villageplumbing.com    TX   
5008  Real Restoration Group        realrestoration.com    IL   
5009           Naval Systems                   n-s-i.us    MD   
5010             HNM Systems             hnmsystems.com    CA   
5011                 Vivayic                vivayic.com    NE   

      revenue in Millions    growth_%                      industry  workers  \
0                    36.9  36680.3882       Advertising & Marketing     40.0   
1                    33.6  30547.9317    Logistics & Transportation     39.0   
2                    24.9  23880.4852               Food & Beverage    190.0   
3                    32.4  21849.8925  Consumer Products & Services     57.0   
4                    22.5  18166.4070                        Retail     25.0   
...                   ...         ...                           ...      ...   
5007                 15.8     52.2377  Consumer Products & Services     88.0   
5008                 11.6     52.2127                  Construction    380.0   
5009                 29.7     52.2037           Government Services    187.0   
5010                  8.8     52.1919            Telecommunications    132.0   
5011                  4.5     52.1691  Business Products & Services     27.0   

      previous_workers  founded  yrs_on_list         metro            city  \
0                    5     2015            1       Phoenix         Phoenix   
1                    8     2015            1     Nashville       Brentwood   
2                   10     2015            1        Austin          Austin   
3                    2     2014            1           NaN     Albuquerque   
4                    6     2014            1  Philadelphia    Philadelphia   
...                ...      ...          ...           ...             ...   
5007                62     1946            3       Houston         Houston   
5008               220     2011            1       Chicago         Chicago   
5009               127     2004            1           NaN  LEXINGTON PARK   
5010                47     2011            1     San Diego    Solana Beach   
5011                22     2006            4           NaN         Lincoln   

         Unit  
0     Million  
1     Million  
2     Million  
3     Million  
4     Million  
...       ...  
5007  Million  
5008  Million  
5009  Million  
5010  Million  
5011  Million  

[5012 rows x 15 columns]

In [17]:
# dropping out the field "Unit" as we don't need it anymore.
avg_revenue = avg_revenue.drop("Unit" , axis = 1)

In [18]:
avg_revenue

rank                                            profile  \
0        1               https://www.inc.com/profile/freestar   
1        2            https://www.inc.com/profile/freightwise   
2        3           https://www.inc.com/profile/ceces-veggie   
3        4               https://www.inc.com/profile/ladyboss   
4        5                 https://www.inc.com/profile/perpay   
...    ...                                                ...   
5007  4996   https://www.inc.com/profile/village-plumbing-air   
5008  4997  https://www.inc.com/profile/real-restoration-g...   
5009  4998          https://www.inc.com/profile/naval-systems   
5010  4999            https://www.inc.com/profile/hnm-systems   
5011  5000                https://www.inc.com/profile/vivayic   

                        name                        url state  \
0                   Freestar        http://freestar.com    AZ   
1                FreightWise  http://freightwisellc.com    TN   
2          Cece's Veggie Co.   http://cecesveggieco.com    TX   
3                   LadyBoss        http://ladyboss.com    NM   
4                     Perpay          http://perpay.com    PA   
...                      ...                        ...   ...   
5007  Village Plumbing & Air        villageplumbing.com    TX   
5008  Real Restoration Group        realrestoration.com    IL   
5009           Naval Systems                   n-s-i.us    MD   
5010             HNM Systems             hnmsystems.com    CA   
5011                 Vivayic                vivayic.com    NE   

      revenue in Millions    growth_%                      industry  workers  \
0                    36.9  36680.3882       Advertising & Marketing     40.0   
1                    33.6  30547.9317    Logistics & Transportation     39.0   
2                    24.9  23880.4852               Food & Beverage    190.0   
3                    32.4  21849.8925  Consumer Products & Services     57.0   
4                    22.5  18166.4070                        Retail     25.0   
...                   ...         ...                           ...      ...   
5007                 15.8     52.2377  Consumer Products & Services     88.0   
5008                 11.6     52.2127                  Construction    380.0   
5009                 29.7     52.2037           Government Services    187.0   
5010                  8.8     52.1919            Telecommunications    132.0   
5011                  4.5     52.1691  Business Products & Services     27.0   

      previous_workers  founded  yrs_on_list         metro            city  
0                    5     2015            1       Phoenix         Phoenix  
1                    8     2015            1     Nashville       Brentwood  
2                   10     2015            1        Austin          Austin  
3                    2     2014            1           NaN     Albuquerque  
4                    6     2014            1  Philadelphia    Philadelphia  
...                ...      ...          ...           ...             ...  
5007                62     1946            3       Houston         Houston  
5008               220     2011            1       Chicago         Chicago  
5009               127     2004            1           NaN  LEXINGTON PARK  
5010                47     2011            1     San Diego    Solana Beach  
5011                22     2006            4           NaN         Lincoln  

[5012 rows x 14 columns]

In [19]:
avg_revenue.columns

Index(['rank', 'profile', 'name', 'url', 'state', 'revenue in Millions',
       'growth_%', 'industry', 'workers', 'previous_workers', 'founded',
       'yrs_on_list', 'metro', 'city'],
      dtype='object')

# **1. What's the average revenue among companies on the list?**

In [20]:
avg_revenue["revenue in Millions"].describe()

count     5012.000000
mean        47.470830
std        391.334265
min          2.000000
25%          4.800000
50%         10.400000
75%         27.000000
max      21400.000000
Name: revenue in Millions, dtype: float64

## Average revenue among the companies on the list is 47.47 Millions.

# **1. What's the average revenue among companies on the list Broken down by industry?**

 to determine the average revenue generated by companies we have to group the companies and then measure its average/mean

In [21]:
# grouping wrt industry and measuring its average/mean
revenue_among_companies = avg_revenue.groupby(by= "industry").mean()

In [22]:
# only withdrawing the fields which is required for our analysis.
industries = revenue_among_companies["revenue in Millions"].round(2).sort_values(ascending= False)

In [23]:
industries

industry
Security                        123.75
Health                          109.33
Logistics & Transportation      101.29
Energy                           80.22
Human Resources                  73.58
Consumer Products & Services     72.07
Retail                           66.55
Government Services              59.39
Insurance                        58.74
Construction                     58.42
Telecommunications               41.83
Financial Services               40.00
Food & Beverage                  39.86
Real Estate                      34.06
Manufacturing                    33.30
Engineering                      32.95
Computer Hardware                31.18
Business Products & Services     28.60
Travel & Hospitality             26.74
IT Management                    22.58
Environmental Services           22.51
Software                         21.57
IT System Development            21.09
Education                        20.84
IT Services                      19.13
Advertising & Ma

# **2.Which industries are most and least represented in the list?**

In [24]:
industries

industry
Security                        123.75
Health                          109.33
Logistics & Transportation      101.29
Energy                           80.22
Human Resources                  73.58
Consumer Products & Services     72.07
Retail                           66.55
Government Services              59.39
Insurance                        58.74
Construction                     58.42
Telecommunications               41.83
Financial Services               40.00
Food & Beverage                  39.86
Real Estate                      34.06
Manufacturing                    33.30
Engineering                      32.95
Computer Hardware                31.18
Business Products & Services     28.60
Travel & Hospitality             26.74
IT Management                    22.58
Environmental Services           22.51
Software                         21.57
IT System Development            21.09
Education                        20.84
IT Services                      19.13
Advertising & Ma

### 1. Most represented industry is "Security" with revenue generated around 125 Million  
### 2. Least represented industry is "Media" with revenue generated around 16.7 Million

# **3. Do you notice any interesting geographic trends?**

* ##  Divide the whole data of states into zones and then find out the trends

In [25]:
avg_revenue.head(2)

rank                                  profile         name  \
0     1     https://www.inc.com/profile/freestar     Freestar   
1     2  https://www.inc.com/profile/freightwise  FreightWise   

                         url state  revenue in Millions    growth_%  \
0        http://freestar.com    AZ                 36.9  36680.3882   
1  http://freightwisellc.com    TN                 33.6  30547.9317   

                     industry  workers  previous_workers  founded  \
0     Advertising & Marketing     40.0                 5     2015   
1  Logistics & Transportation     39.0                 8     2015   

   yrs_on_list      metro       city  
0            1    Phoenix    Phoenix  
1            1  Nashville  Brentwood

In [26]:
import plotly.express as px                                             # importing plotpy liebrary
state_counts = avg_revenue["state"].value_counts().reset_index()        # checking value count of the statesrepresenting no. of companies as well.
state_counts.columns = ["state" , "count"]                              # renaming the fields

fig = px.choropleth(state_counts, locationmode = "USA-states", locations = "state" , color = "count",
                   scope = "usa" , title = "Number of Companies by State")              # using choropleth plot for map plot
fig.show()

## With the above fig. we can observe that the highest number of companies in Inc 5000, are in the state of California (CA).California lies in West Region.

In [27]:
states = avg_revenue.groupby(by = "state").mean().reset_index().round(2)
states

state     rank  revenue in Millions  growth_%  workers  previous_workers  \
0     AL  2385.28                37.11    377.78   256.03            130.72   
1     AR  2191.90                11.38    246.40    92.10             32.60   
2     AZ  2538.46                23.14    702.49   144.26             49.59   
3     CA  2390.91                53.14    490.90   549.96            233.00   
4     CO  2460.54                29.25    434.65   153.78             71.16   
5     CT  2677.57                46.80    293.72   132.73             37.62   
6     DC  2376.65                19.00    376.16    94.53             31.14   
7     DE  2238.27                30.96    250.95   200.27             97.27   
8     FL  2379.25                51.41    472.75   200.92             95.16   
9     GA  2499.95                42.36    438.37   176.04             89.14   
10    HI  2321.75                13.00    173.53    21.25              6.00   
11    IA  2821.55                19.86    215.69   334.52            130.74   
12    ID  1414.90                24.88   1335.74    45.71             19.62   
13    IL  2637.61                42.98    350.06   234.21            130.51   
14    IN  2646.81                34.47    299.71   147.21             84.46   
15    KS  2953.22                44.23    416.27   630.41            322.85   
16    KY  2845.82                74.59    537.79   302.06            168.50   
17    LA  2498.57                20.01    310.78    73.76             35.54   
18    MA  2701.65                38.96    284.14   183.19             95.99   
19    MD  2257.48                27.71    553.96   575.88            220.78   
20    ME  1819.82                14.17    686.62   125.45             38.73   
21    MI  2750.60                37.54    517.23   106.84             63.54   
22    MN  2845.05               285.63    322.49   154.77            104.59   
23    MO  2599.55                33.76    319.41   121.44             52.96   
24    MS  2955.75                 6.07    143.74    34.50             17.50   
25    MT  2613.64                 6.42    830.76    40.09             14.27   
26    NC  2347.16                56.65    419.88   206.46            109.70   
27    ND  1814.40                15.05    655.67    51.60             21.10   
28    NE  2728.41                50.99    290.53   157.44             67.52   
29    NH  2645.40                22.78    292.09    57.50             28.85   
30    NJ  2605.91                50.70    447.13   244.30            133.19   
31    NM  2345.71                28.77   3259.42   162.00             74.57   
32    NV  2295.15                20.10    338.12   142.70             73.27   
33    NY  2262.01                27.69    600.11   152.02             68.93   
34    OH  2769.24                79.51    358.71   211.88            110.88   
35    OK  2649.71                24.96    219.16   122.82             59.07   
36    OR  2665.44                58.68    229.44   143.53             68.95   
37    PA  2855.50                64.03    401.73   184.07            104.01   
38    PR   422.50                 5.35   1109.55    18.50              7.00   
39    RI  3724.00                 6.62     95.28    38.60             20.80   
40    SC  2387.50                26.00    311.49   101.71             47.61   
41    SD  2724.07                54.01    192.12    71.00             38.21   
42    TN  2728.38                40.94    734.20   247.25            122.10   
43    TX  2516.45                42.89    433.90   170.07             83.12   
44    UT  2306.77                55.20    449.36   327.89             99.37   
45    VA  2513.44                44.03    472.69   138.29             61.57   
46    VT  3016.80                 6.48    139.86    46.00             21.60   
47    WA  2351.39                27.86    531.75   163.50             75.85   
48    WI  2859.84                25.72    456.11    86.10             45.04   
49    WV  2010.43                24.91    223.13    9

In [28]:
state_revenue = states[["state" , "revenue in Millions"]]        # checking value count of the statesrepresenting no. of companies as well.
state_revenue.columns = ["state" , "revenue in Millions"]                              # renaming the fields

fig = px.choropleth(state_revenue, locationmode = "USA-states", locations = "state" , color ="revenue in Millions",
                    scope = "usa" , title = "Revenue Generated in differrent State")              # using choropleth plot for map plot
fig.show()

## With the above fig, we can observe that the highest revenue generated by companies are situated in Minnesota (MN) with revenue generate around 285 millions. 

In [29]:
lst = [avg_revenue.state.unique()]
lst

[array(['AZ', 'TN', 'TX', 'NM', 'PA', 'FL', 'NJ', 'VA', 'OH', 'CA', 'CO',
        'WA', 'NY', 'GA', 'KY', 'ID', 'UT', 'MT', 'NC', 'WI', 'MI', 'MD',
        'MN', 'AL', 'MA', 'ME', 'KS', 'IL', 'CT', 'NV', 'ND', 'NH', 'MO',
        'IN', 'DC', 'NE', 'WY', 'SC', 'LA', 'PR', 'OR', 'IA', 'OK', 'AR',
        'DE', 'SD', 'WV', 'MS', 'VT', 'HI', 'RI'], dtype=object)]

In [30]:
west = ["CA","CO", "HI", "ID", "MT", "NV", "OR", "UT", "WA", "WY"]
midwest = ['OH','WI', 'MI','MN','KS', 'IL','ND','MO','IN','NE', 'IA','SD']
northeast =["CT", 'DC', 'DE', 'MA','MD','ME','NH','NJ','NY','PA','RI','VT']
southeast = ['AL','AR','FL', 'GA', 'KY', 'LA', 'MS', 'NC', 'PR','SC','TN', 'VA', 'WV']
southwest = ['AZ', 'NM', 'OK', 'TX']

In [31]:
def categorise(row):  
    if row['state'] in west :
        return 'west'
    elif row['state'] in midwest :
        return 'midwest'
    elif row['state'] in northeast :
        return 'northeast'
    elif row['state'] in southeast :
        return 'southeast'
    elif row['state'] in southwest :
        return 'southwest'
    
    
avg_revenue['zone'] = avg_revenue.apply(lambda row: categorise(row), axis=1)
avg_revenue

rank                                            profile  \
0        1               https://www.inc.com/profile/freestar   
1        2            https://www.inc.com/profile/freightwise   
2        3           https://www.inc.com/profile/ceces-veggie   
3        4               https://www.inc.com/profile/ladyboss   
4        5                 https://www.inc.com/profile/perpay   
...    ...                                                ...   
5007  4996   https://www.inc.com/profile/village-plumbing-air   
5008  4997  https://www.inc.com/profile/real-restoration-g...   
5009  4998          https://www.inc.com/profile/naval-systems   
5010  4999            https://www.inc.com/profile/hnm-systems   
5011  5000                https://www.inc.com/profile/vivayic   

                        name                        url state  \
0                   Freestar        http://freestar.com    AZ   
1                FreightWise  http://freightwisellc.com    TN   
2          Cece's Veggie Co.   http://cecesveggieco.com    TX   
3                   LadyBoss        http://ladyboss.com    NM   
4                     Perpay          http://perpay.com    PA   
...                      ...                        ...   ...   
5007  Village Plumbing & Air        villageplumbing.com    TX   
5008  Real Restoration Group        realrestoration.com    IL   
5009           Naval Systems                   n-s-i.us    MD   
5010             HNM Systems             hnmsystems.com    CA   
5011                 Vivayic                vivayic.com    NE   

      revenue in Millions    growth_%                      industry  workers  \
0                    36.9  36680.3882       Advertising & Marketing     40.0   
1                    33.6  30547.9317    Logistics & Transportation     39.0   
2                    24.9  23880.4852               Food & Beverage    190.0   
3                    32.4  21849.8925  Consumer Products & Services     57.0   
4                    22.5  18166.4070                        Retail     25.0   
...                   ...         ...                           ...      ...   
5007                 15.8     52.2377  Consumer Products & Services     88.0   
5008                 11.6     52.2127                  Construction    380.0   
5009                 29.7     52.2037           Government Services    187.0   
5010                  8.8     52.1919            Telecommunications    132.0   
5011                  4.5     52.1691  Business Products & Services     27.0   

      previous_workers  founded  yrs_on_list         metro            city  \
0                    5     2015            1       Phoenix         Phoenix   
1                    8     2015            1     Nashville       Brentwood   
2                   10     2015            1        Austin          Austin   
3                    2     2014            1           NaN     Albuquerque   
4                    6     2014            1  Philadelphia    Philadelphia   
...                ...      ...          ...           ...             ...   
5007                62     1946            3       Houston         Houston   
5008               220     2011            1       Chicago         Chicago   
5009               127     2004            1           NaN  LEXINGTON PARK   
5010                47     2011            1     San Diego    Solana Beach   
5011                22     2006            4           NaN         Lincoln   

           zone  
0     southwest  
1     southeast  
2     southwest  
3     southwest  
4     northeast  
...         ...  
5007  southwest  
5008    midwest  
5009  northeast  
5010       west  
5011    midwest  

[5012 rows x 15 columns]

In [32]:
zonewise = avg_revenue.copy()

In [33]:
company_counts = zonewise.groupby(by = 'zone').size()
company_counts

zone
midwest       860
northeast     995
southeast    1325
southwest     606
west         1226
dtype: int64

## Around 1325 companies are in southeast region/zone which is the highest among all the regions/zones.

In [34]:
zonewise_total_revenue = zonewise.groupby(by = 'zone').sum()
zonewise_total_revenue[["revenue in Millions"]]

revenue in Millions
zone                          
midwest                59351.0
northeast              38160.3
southeast              60267.7
southwest              23335.4
west                   56809.4

## Companies in southeast region generates highest total revenue among all the region which is around 60267.7 Millions, followed by midwest region and then west region by generating around 59351 and 56809.4 Millions respectively.

# **4. Which industries saw the largest average growth rate?**

In [35]:
growth_rate = avg_revenue.groupby(by = "industry").mean().reset_index()          # grouping wrt industry and measuring its average/mean
max_growth_rate = growth_rate[["industry" , "growth_%"]].round(2)
max_growth_rate

industry  growth_%
0        Advertising & Marketing    486.69
1   Business Products & Services    313.47
2              Computer Hardware    651.98
3                   Construction    358.09
4   Consumer Products & Services    685.64
5                      Education    384.56
6                         Energy    592.08
7                    Engineering    222.76
8         Environmental Services    458.55
9             Financial Services    423.94
10               Food & Beverage    668.23
11           Government Services    485.31
12                        Health    546.91
13               Human Resources    288.71
14                 IT Management    275.36
15                   IT Services     70.58
16         IT System Development    385.91
17                     Insurance    306.15
18    Logistics & Transportation    717.56
19                 Manufacturing    333.70
20                         Media    413.73
21                   Real Estate    417.55
22                        Retail    584.82
23                      Security    490.19
24                      Software    524.67
25            Telecommunications    515.41
26          Travel & Hospitality    346.42

In [36]:
max_growth_rate.iloc[0]

industry    Advertising & Marketing
growth_%                     486.69
Name: 0, dtype: object

## Advertising & Marketing saw the largest average growth rate of around 486.69%.

# **5. Which companies had the largest increase in staff/new hires?**

* ## Top 5 and Bottom 5 - Companies  

In [37]:
staff_hire = avg_revenue.copy()

In [38]:
staff_hire.head()

rank                                   profile               name  \
0     1      https://www.inc.com/profile/freestar           Freestar   
1     2   https://www.inc.com/profile/freightwise        FreightWise   
2     3  https://www.inc.com/profile/ceces-veggie  Cece's Veggie Co.   
3     4      https://www.inc.com/profile/ladyboss           LadyBoss   
4     5        https://www.inc.com/profile/perpay             Perpay   

                         url state  revenue in Millions    growth_%  \
0        http://freestar.com    AZ                 36.9  36680.3882   
1  http://freightwisellc.com    TN                 33.6  30547.9317   
2   http://cecesveggieco.com    TX                 24.9  23880.4852   
3        http://ladyboss.com    NM                 32.4  21849.8925   
4          http://perpay.com    PA                 22.5  18166.4070   

                       industry  workers  previous_workers  founded  \
0       Advertising & Marketing     40.0                 5     2015   
1    Logistics & Transportation     39.0                 8     2015   
2               Food & Beverage    190.0                10     2015   
3  Consumer Products & Services     57.0                 2     2014   
4                        Retail     25.0                 6     2014   

   yrs_on_list         metro          city       zone  
0            1       Phoenix       Phoenix  southwest  
1            1     Nashville     Brentwood  southeast  
2            1        Austin        Austin  southwest  
3            1           NaN   Albuquerque  southwest  
4            1  Philadelphia  Philadelphia  northeast

In [39]:
staff_hire.columns

Index(['rank', 'profile', 'name', 'url', 'state', 'revenue in Millions',
       'growth_%', 'industry', 'workers', 'previous_workers', 'founded',
       'yrs_on_list', 'metro', 'city', 'zone'],
      dtype='object')

In [40]:
staff_hire["no. of hired employee"] = staff_hire["workers"] - staff_hire["previous_workers"]    # subtracting field previous workers from workers field 
                                                                                                # and assigning values in the new column

In [41]:
staff_hiring_companies = staff_hire.copy()                                      

In [42]:
# Creating new dataset which contains all the positive values in the field of "no. of hired employee" it represent the no. of employee hired by the company.
staff_hiring_companies = staff_hiring_companies[staff_hiring_companies["no. of hired employee"]>0]        

In [43]:
#sorting values on the basis of no. of employee hired.
staff_hiring_companies = staff_hiring_companies.sort_values(by = "no. of hired employee", ascending = False )
staff_hiring_companies

rank                                            profile  \
1621  1617       https://www.inc.com/profile/allied-universal   
3752  3743                https://www.inc.com/profile/alorica   
3403  3394            https://www.inc.com/profile/infiniti-hr   
1336  1333             https://www.inc.com/profile/circustrix   
3172  3164          https://www.inc.com/profile/scribeamerica   
...    ...                                                ...   
411    411     https://www.inc.com/profile/pitisci-associates   
1348  1344  https://www.inc.com/profile/suite-experience-g...   
397    397            https://www.inc.com/profile/double-wood   
811    811           https://www.inc.com/profile/grillaholics   
4646  4635           https://www.inc.com/profile/julianna-rae   

                        name                        url state  \
1621        Allied Universal             http://aus.com    CA   
3752                 Alorica                alorica.com    CA   
3403             Infiniti HR      http://infinitihr.com    MD   
1336              CircusTrix             circustrix.com    UT   
3172           ScribeAmerica          scribeamerica.com    FL   
...                      ...                        ...   ...   
411     Pitisci & Associates             tampastaff.com    FL   
1348  Suite Experience Group   suiteexperiencegroup.com    CA   
397              Double Wood  doublewoodsupplements.com    PA   
811             Grillaholics           grillaholics.com    MO   
4646            Julianna Rae            juliannarae.com    MA   

      revenue in Millions   growth_%                      industry   workers  \
1621               5800.0   255.1902                      Security  155000.0   
3752                  2.2    90.9001  Business Products & Services   96000.0   
3403                458.0   104.7641               Human Resources   58145.0   
1336                194.1   309.4015       Advertising & Marketing   14730.0   
3172                305.1   115.8777                        Health   20599.0   
...                   ...        ...                           ...       ...   
411                   8.8  1110.7164                 IT Management       7.0   
1348                 10.6   307.4624          Travel & Hospitality       4.0   
397                   6.9  1147.2142                        Health       3.0   
811                   2.7   527.2026                        Retail       3.0   
4646                  2.1    61.9375                        Retail       5.0   

      previous_workers  founded  yrs_on_list             metro  \
1621             36469     1957           10       Los Angeles   
3752             53000     1999            2       Los Angeles   
3403             21000     2008            7    Washington, DC   
1336               991     2011            3    Provo-Orem, UT   
3172              8958     2005            9             Miami   
...                ...      ...          ...               ...   
411                  6     2014            1             Tampa   
1348                 3     2010            2     San Francisco   
397                  2     2013            1      Philadelphia   
811                  2     2014            1  St. Louis, MO-IL   
4646                 4     2004            1            Boston   

                 city       zone  no. of hired employee  
1621        Santa Ana       west               118531.0  
3752           Irvine       west                43000.0  
3403     Burtonsville  northeast                37145.0  
1336            Provo       west                13739.0  
3172  Fort Lauderdale  southeast                11641.0  
...               ...        ...                    ...  
411             Tampa  southeast                    1.0  
1348        San Mateo       west                    1.0  
397      Philadelphia  northeast                    1.0  
811         New Melle    midwest                    1.0  
4646       Burlington  northeast                    

## **Top 5 companies with the largest increase in staff/new hires**

In [44]:
Top_5_companies = staff_hiring_companies[["name" , "no. of hired employee"]].head()
Top_5_companies

name  no. of hired employee
1621  Allied Universal               118531.0
3752           Alorica                43000.0
3403       Infiniti HR                37145.0
1336        CircusTrix                13739.0
3172     ScribeAmerica                11641.0

## **Bottom 5 companies with the lowest increase in staff/new hires**

In [45]:
Bottom_5_companies = staff_hiring_companies[["name" , "no. of hired employee"]].tail()
Bottom_5_companies

name  no. of hired employee
411     Pitisci & Associates                    1.0
1348  Suite Experience Group                    1.0
397              Double Wood                    1.0
811             Grillaholics                    1.0
4646            Julianna Rae                    1.0

# **6. Did any companies increase revenue while reducing staff?**

In [46]:
companies_reducing_staff = staff_hire.copy()

In [47]:
# Creating new dataset which contains all the negative values in the field of "no. of hired employee" it represent the no. of employee laid off by the company.
companies_reducing_staff = companies_reducing_staff[companies_reducing_staff["no. of hired employee"]<0]
companies_reducing_staff

rank                                            profile  \
245    245            https://www.inc.com/profile/tier4-group   
361    361              https://www.inc.com/profile/adficient   
475    475  https://www.inc.com/profile/powerful-professio...   
525    525      https://www.inc.com/profile/quantico-tactical   
534    534                  https://www.inc.com/profile/peaxy   
...    ...                                                ...   
4867  4856                https://www.inc.com/profile/openarc   
4942  4931  https://www.inc.com/profile/green-mountain-tec...   
4979  4968      https://www.inc.com/profile/atlas-oil-company   
5001  4990     https://www.inc.com/profile/smart-energy-today   
5002  4991             https://www.inc.com/profile/Distillery   

                           name                               url state  \
245                 Tier4 Group             http://tier4group.com    GA   
361                   AdFicient                     adficient.com    MO   
475      Powerful Professionals  http://powerfulprofessionals.com    GA   
525           Quantico Tactical              quanticotactical.com    NC   
534                       Peaxy                         peaxy.net    CA   
...                         ...                               ...   ...   
4867                    OpenArc                       openarc.net    PA   
4942  Green Mountain Technology       greenmountaintechnology.com    TN   
4979          Atlas Oil Company                      atlasoil.com    MI   
5001         Smart Energy Today              smartenergytoday.net    WA   
5002                 Distillery             http://distillery.com    CA   

      revenue in Millions   growth_%                      industry  workers  \
245                   4.6  1729.3981  Business Products & Services      0.0   
361                   5.9  1264.9067       Advertising & Marketing      3.0   
475                   2.0   935.0348  Business Products & Services      5.0   
525                 492.5   831.3181           Government Services     37.0   
534                   2.9   817.2645                      Software     20.0   
...                   ...        ...                           ...      ...   
4867                 15.0    56.7225                 IT Management     90.0   
4942                 21.6    54.2047    Logistics & Transportation      0.0   
4979               1100.0    53.0161                        Energy    538.0   
5001                 20.5    52.5381                        Energy     65.0   
5002                  6.1    52.5232                      Software     11.0   

      previous_workers  founded  yrs_on_list             metro          city  \
245                  3     2010            1           Atlanta       Atlanta   
361                  4     2009            1  St. Louis, MO-IL   Saint Louis   
475                 13     2001            1           Atlanta    Alpharetta   
525                 80     2001            2               NaN      Aberdeen   
534                 46     2012            2          San Jose      San Jose   
...                ...      ...          ...               ...           ...   
4867                95     2010            5    Pittsburgh, PA       Wexford   
4942                48     1999            3               NaN       Memphis   
4979               611     1985            6           Detroit        Taylor   
5001                92     2012            6               NaN      Tumwater   
5002                13     2012            3       Los Angeles  Santa Monica   

           zone  no. of hired employee  
245   southeast                   -3.0  
361     midwest                   -1.0  
475   southeast                   -8.0  
525   southeast                  -43.0  
534        west                  -26.0  
...         ...                    ...  
4867  northeast                   -5.0  
4942  southeast                  -48.0  
4979    midwest                  -73.0  
5001

In [48]:
#sorting values on the basis of no. of employee hired.
companies_reducing_staff = companies_reducing_staff.sort_values(by = "no. of hired employee", ascending = True)
companies_reducing_staff

rank                                            profile  \
4567  4556  https://www.inc.com/profile/ids-international-...   
1076  1073     https://www.inc.com/profile/prime-therapeutics   
4409  4398                 https://www.inc.com/profile/akorbi   
4136  4127  https://www.inc.com/profile/first-american-pay...   
4270  4260     https://www.inc.com/profile/integrity-services   
...    ...                                                ...   
2966  2958          https://www.inc.com/profile/alpha-premier   
3194  3186             https://www.inc.com/profile/rabbit-air   
3799  3790          https://www.inc.com/profile/cuddle-clones   
4657  4646           https://www.inc.com/profile/klemchuk-llp   
4258  4248      https://www.inc.com/profile/sbs-cybersecurity   

                                       name                            url  \
4567  IDS International Government Services           idsinternational.com   
1076                     Prime Therapeutics   http://primetherapeutics.com   
4409                                 Akorbi                     akorbi.com   
4136         First American Payment Systems             first-american.net   
4270                     Integrity Services  integrityservicecompanies.com   
...                                     ...                            ...   
2966                          Alpha Premier          alpharoofingtexas.com   
3194                             Rabbit Air                  rabbitair.com   
3799                          Cuddle Clones               cuddleclones.com   
4657                               Klemchuk            http://klemchuk.com   
4258                      SBS CyberSecurity                   sbscyber.com   

     state  revenue in Millions  growth_%                      industry  \
4567    VA                 44.7   64.0436           Government Services   
1076    MN              21400.0  390.5654                        Health   
4409    TX                 39.3   68.1185  Business Products & Services   
4136    TX                783.9   77.1276            Financial Services   
4270    MA                 19.3   72.2406  Business Products & Services   
...    ...                  ...       ...                           ...   
2966    TX                  2.6  126.3750                  Construction   
3194    CA                 18.4  114.9680  Consumer Products & Services   
3799    KY                  3.5   88.4597  Consumer Products & Services   
4657    TX                  4.5   61.6037  Business Products & Services   
4258    SD                 10.1   72.5545            Financial Services   

      workers  previous_workers  founded  yrs_on_list  \
4567      0.0               682     2006            3   
1076   3300.0              3900     1998            8   
4409    350.0               650     2003            4   
4136   1001.0              1244     1990            2   
4270    100.0               250     2008            6   
...       ...               ...      ...          ...   
2966      4.0                 5     2014            1   
3194     19.0                20     2004            7   
3799     50.0                51     2011            2   
4657     17.0                18     2009            1   
4258     71.0                72     2004            7   

                                   metro        city       zone  \
4567                      Washington, DC   Arlington  southeast   
1076                         Minneapolis       Eagan    midwest   
4409                              Dallas       Plano  southwest   
4136                              Dallas  Fort Worth  southwest   
4270                              Boston   Wakefield  northeast   
...                                  ...         ...        ...   
2966                              Austin  Round Rock  southwest   
3194                         Los Angeles    Pasadena       west   
3799  Louisville/Jefferson County, KY-IN  Louisville  southeast   
4657                            

In [49]:
companies_reducing_staff.head(10)

rank                                            profile  \
4567  4556  https://www.inc.com/profile/ids-international-...   
1076  1073     https://www.inc.com/profile/prime-therapeutics   
4409  4398                 https://www.inc.com/profile/akorbi   
4136  4127  https://www.inc.com/profile/first-american-pay...   
4270  4260     https://www.inc.com/profile/integrity-services   
4153  4144               https://www.inc.com/profile/centrify   
3627  3618                   https://www.inc.com/profile/zefr   
4583  4572                  https://www.inc.com/profile/ctems   
4979  4968      https://www.inc.com/profile/atlas-oil-company   
2715  2707                   https://www.inc.com/profile/ncwc   

                                       name                            url  \
4567  IDS International Government Services           idsinternational.com   
1076                     Prime Therapeutics   http://primetherapeutics.com   
4409                                 Akorbi                     akorbi.com   
4136         First American Payment Systems             first-american.net   
4270                     Integrity Services  integrityservicecompanies.com   
4153                               Centrify                   centrify.com   
3627                                   Zefr                       zefr.com   
4583                                  Ctems           http://ctems-inc.com   
4979                      Atlas Oil Company                   atlasoil.com   
2715                                   NCWC                    ncwcinc.com   

     state  revenue in Millions  growth_%                      industry  \
4567    VA                 44.7   64.0436           Government Services   
1076    MN              21400.0  390.5654                        Health   
4409    TX                 39.3   68.1185  Business Products & Services   
4136    TX                783.9   77.1276            Financial Services   
4270    MA                 19.3   72.2406  Business Products & Services   
4153    CA                114.6   76.4500                      Software   
3627    CA                 91.8   96.5302       Advertising & Marketing   
4583    CA                 15.2   63.6505                 Manufacturing   
4979    MI               1100.0   53.0161                        Energy   
2715    NJ                195.8  142.2609       Advertising & Marketing   

      workers  previous_workers  founded  yrs_on_list           metro  \
4567      0.0               682     2006            3  Washington, DC   
1076   3300.0              3900     1998            8     Minneapolis   
4409    350.0               650     2003            4          Dallas   
4136   1001.0              1244     1990            2          Dallas   
4270    100.0               250     2008            6          Boston   
4153    211.0               338     2004            6        San Jose   
3627    227.0               313     2008            1     Los Angeles   
4583     70.0               150     2011            4        San Jose   
4979    538.0               611     1985            6         Detroit   
2715     54.0               125     1993            6   New York City   

             city       zone  no. of hired employee  
4567    Arlington  southeast                 -682.0  
1076        Eagan    midwest                 -600.0  
4409        Plano  southwest                 -300.0  
4136   Fort Worth  southwest                 -243.0  
4270    Wakefield  northeast                 -150.0  
4153  Santa Clara       west                 -127.0  
3627  Los Angeles       west                  -86.0  
4583     Milpitas       west                  -80.0  
4979       Taylor    midwest                  -73.0  
2715        Ocean  northeast                  -71.0

In [50]:
# using the required fields only
top_10_companies_reducing_staff = companies_reducing_staff[["name","no. of hired employee", "growth_%"]].head(10)
top_10_companies_reducing_staff

name  no. of hired employee  growth_%
4567  IDS International Government Services                 -682.0   64.0436
1076                     Prime Therapeutics                 -600.0  390.5654
4409                                 Akorbi                 -300.0   68.1185
4136         First American Payment Systems                 -243.0   77.1276
4270                     Integrity Services                 -150.0   72.2406
4153                               Centrify                 -127.0   76.4500
3627                                   Zefr                  -86.0   96.5302
4583                                  Ctems                  -80.0   63.6505
4979                      Atlas Oil Company                  -73.0   53.0161
2715                                   NCWC                  -71.0  142.2609

## Top 10 companies which manages to improve growth % even after reducing the working staff.
1. IDS International Government Services
2. Prime Therapeutics
3. Akorbi
4. First American Payment Systems
5. Integrity Services
6. Centrify
7. Zefr
8. Ctems
9. Atlas Oil Company
10. NCWC


# **7. Top 1000 - Winner State**

In [51]:
top_1000 = avg_revenue.copy()

In [52]:
top_1000 = top_1000[top_1000["rank"]<=1000]                          # creating new dataset for top 1000 ranked companies
top_1000

rank                                            profile  \
0        1               https://www.inc.com/profile/freestar   
1        2            https://www.inc.com/profile/freightwise   
2        3           https://www.inc.com/profile/ceces-veggie   
3        4               https://www.inc.com/profile/ladyboss   
4        5                 https://www.inc.com/profile/perpay   
...    ...                                                ...   
997    996       https://www.inc.com/profile/farmgirl-flowers   
998    997                https://www.inc.com/profile/bombora   
999    998        https://www.inc.com/profile/frugal-flippers   
1000   999    https://www.inc.com/profile/sumo-communications   
1001  1000  https://www.inc.com/profile/home-view-technolo...   

                        name                        url state  \
0                   Freestar        http://freestar.com    AZ   
1                FreightWise  http://freightwisellc.com    TN   
2          Cece's Veggie Co.   http://cecesveggieco.com    TX   
3                   LadyBoss        http://ladyboss.com    NM   
4                     Perpay          http://perpay.com    PA   
...                      ...                        ...   ...   
997         Farmgirl Flowers        farmgirlflowers.com    CA   
998                  Bombora                bombora.com    NY   
999          Frugal Flippers        householdtrends.com    NY   
1000     SUMO Communications                sumocom.com    UT   
1001  Home View Technologies           liveviewtech.com    UT   

      revenue in Millions    growth_%                      industry  workers  \
0                    36.9  36680.3882       Advertising & Marketing     40.0   
1                    33.6  30547.9317    Logistics & Transportation     39.0   
2                    24.9  23880.4852               Food & Beverage    190.0   
3                    32.4  21849.8925  Consumer Products & Services     57.0   
4                    22.5  18166.4070                        Retail     25.0   
...                   ...         ...                           ...      ...   
997                  23.4    424.8674                        Retail    106.0   
998                  26.4    424.7169       Advertising & Marketing     95.0   
999                   5.6    424.6089                        Retail      5.0   
1000                  5.1    424.4421            Telecommunications     12.0   
1001                 16.5    424.0165                      Security    110.0   

      previous_workers  founded  yrs_on_list                 metro  \
0                    5     2015            1               Phoenix   
1                    8     2015            1             Nashville   
2                   10     2015            1                Austin   
3                    2     2014            1                   NaN   
4                    6     2014            1          Philadelphia   
...                ...      ...          ...                   ...   
997                 41     2010            2         San Francisco   
998                 19     2014            1         New York City   
999                  1     2015            1                   NaN   
1000                 2     2002            2  Ogden-Clearfield, UT   
1001                30     2005            1        Provo-Orem, UT   

               city       zone  
0           Phoenix  southwest  
1         Brentwood  southeast  
2            Austin  southwest  
3       Albuquerque  southwest  
4      Philadelphia  northeast  
...             ...        ...  
997   San Francisco       west  
998        New York  northeast  
999          Jordan  northeast  
1000    Centerville       west  
1001           Orem       west  

[1002 rows x 15 columns]

In [53]:
#State wise company count in top 1000 companies.
winner_state = top_1000.groupby(by = "state").size().sort_values(ascending = False)

In [54]:
winner_state

state
CA    170
TX     82
NY     79
FL     78
VA     62
GA     44
IL     43
MD     36
CO     34
PA     28
UT     28
NC     26
WA     23
NJ     22
OH     21
MI     19
AZ     18
MA     16
TN     15
IN     13
MO     13
SC     12
ID     11
AL     11
MN     11
DC     10
LA      9
WI      8
NE      6
NV      6
OR      6
CT      6
KY      5
KS      5
ME      4
ND      3
IA      3
MT      2
PR      2
OK      2
NH      2
AR      2
DE      2
WY      2
SD      1
NM      1
dtype: int64

## Winner state which has highest no. of companies in top 1000 ranked companies is California (CA).

# **8.Do we have any company which was founded last year and Ranked under Top 100**

In [55]:
company_ranked_within_top_100 = avg_revenue.copy()

In [56]:
# creating a dataset which has top 100 ranked companies
company_ranked_within_top_100 = company_ranked_within_top_100[company_ranked_within_top_100["rank"]<=100]
company_ranked_within_top_100

rank                                            profile  \
0       1               https://www.inc.com/profile/freestar   
1       2            https://www.inc.com/profile/freightwise   
2       3           https://www.inc.com/profile/ceces-veggie   
3       4               https://www.inc.com/profile/ladyboss   
4       5                 https://www.inc.com/profile/perpay   
..    ...                                                ...   
96     96         https://www.inc.com/profile/amenity-health   
97     97             https://www.inc.com/profile/payability   
98     98  https://www.inc.com/profile/eagle-transport-an...   
99     99                https://www.inc.com/profile/shipbob   
100   100        https://www.inc.com/profile/powerhome-solar   

                              name                        url state  \
0                         Freestar        http://freestar.com    AZ   
1                      FreightWise  http://freightwisellc.com    TN   
2                Cece's Veggie Co.   http://cecesveggieco.com    TX   
3                         LadyBoss        http://ladyboss.com    NM   
4                           Perpay          http://perpay.com    PA   
..                             ...                        ...   ...   
96                  Amenity Health          amenityhealth.com    CA   
97                      Payability      http://payability.com    NY   
98   Eagle Transport and Logistics      theeagletransport.com    OH   
99                         ShipBob                shipbob.com    IL   
100                Powerhome Solar       http://powerhome.com    NC   

     revenue in Millions    growth_%                      industry  workers  \
0                   36.9  36680.3882       Advertising & Marketing     40.0   
1                   33.6  30547.9317    Logistics & Transportation     39.0   
2                   24.9  23880.4852               Food & Beverage    190.0   
3                   32.4  21849.8925  Consumer Products & Services     57.0   
4                   22.5  18166.4070                        Retail     25.0   
..                   ...         ...                           ...      ...   
96                  15.3   3357.7038                        Health     16.0   
97                  13.7   3350.0091            Financial Services     54.0   
98                   4.8   3339.6237    Logistics & Transportation     35.0   
99                  34.2   3325.7138    Logistics & Transportation    450.0   
100                102.9   3307.1657                  Construction    585.0   

     previous_workers  founded  yrs_on_list          metro          city  \
0                   5     2015            1        Phoenix       Phoenix   
1                   8     2015            1      Nashville     Brentwood   
2                  10     2015            1         Austin        Austin   
3                   2     2014            1            NaN   Albuquerque   
4                   6     2014            1   Philadelphia  Philadelphia   
..                ...      ...          ...            ...           ...   
96                  8     2011            1      San Diego     San Diego   
97                 10     2014            1  New York City      New York   
98                  1     2014            1   Columbus, OH      Columbus   
99                 25     2014            1        Chicago       Chicago   
100                15     2015            2            NaN   Mooresville   

          zone  
0    southwest  
1    southeast  
2    southwest  
3    southwest  
4    northeast  
..         ...  
96        west  
97   northeast  
98     midwest  
99     midwest  
100  southeast  

[101 rows x 15 columns]

In [57]:
# creating a dataset which has companies founded within 1 year i.e in 2015 and 2016
company_ranked_within_top_100 = company_ranked_within_top_100[company_ranked_within_top_100["founded"] >= 2015 ]

In [58]:
company_ranked_within_top_100

rank                                            profile  \
0       1               https://www.inc.com/profile/freestar   
1       2            https://www.inc.com/profile/freightwise   
2       3           https://www.inc.com/profile/ceces-veggie   
7       8  https://www.inc.com/profile/connected-solution...   
13     14            https://www.inc.com/profile/yieldstreet   
15     16       https://www.inc.com/profile/anchor-worldwide   
19     20        https://www.inc.com/profile/kindred-bravely   
20     21  https://www.inc.com/profile/tailor-made-compou...   
23     24               https://www.inc.com/profile/printify   
24     25                https://www.inc.com/profile/awardco   
25     26  https://www.inc.com/profile/pegasus-optimizati...   
29     30               https://www.inc.com/profile/bendcare   
54     55              https://www.inc.com/profile/dropified   
57     58  https://www.inc.com/profile/ashh-dba-ooze-whol...   
65     66       https://www.inc.com/profile/citylift-parking   
71     72  https://www.inc.com/profile/sky-marketing-corp...   
76     77     https://www.inc.com/profile/patton-warehousing   
79     80                 https://www.inc.com/profile/wandrd   
81     82       https://www.inc.com/profile/dynamic-logistix   
85     85               https://www.inc.com/profile/loanatik   
86     86           https://www.inc.com/profile/supportninja   
88     88              https://www.inc.com/profile/cisoshare   
89     89            https://www.inc.com/profile/digitalonus   
100   100        https://www.inc.com/profile/powerhome-solar   

                              name                               url state  \
0                         Freestar               http://freestar.com    AZ   
1                      FreightWise         http://freightwisellc.com    TN   
2                Cece's Veggie Co.          http://cecesveggieco.com    TX   
7        Connected Solutions Group               http://csgstore.net    VA   
13                     YieldStreet            http://yieldstreet.com    NY   
15                Anchor Worldwide                      anchorww.com    NY   
19                 Kindred Bravely         http://kindredbravely.com    CA   
20         Tailor Made Compounding  http://tailormadecompounding.com    KY   
23                        Printify               http://printify.com    CA   
24                         Awardco                   http://award.co    UT   
25   Pegasus Optimization Managers                     pegasus15.com    TX   
29                        Bendcare                      bendcare.com    FL   
54                       Dropified                     dropified.com    AL   
57                  Ooze Wholesale          http://oozewholesale.com    MI   
65                CityLift Parking               cityliftparking.com    CA   
71       Sky Marketing Corporation               smcdistribution.com    TX   
76              Patton Warehousing             pattonwarehousing.com    PA   
79                          Wandrd                 http://wandrd.com    UT   
81                Dynamic Logistix               dynamiclogistix.com    KS   
85                        Loanatik                      loanatik.com    AZ   
86                    SupportNinja           http://supportninja.com    TX   
88                       Cisoshare              http://cisoshare.com    CA   
89                     DigitalOnUS                   digitalonus.com    CA   
100                Powerhome Solar              http://powerhome.com    NC   

     revenue in Millions    growth_%                      industry  workers  \
0                   36.9  36680.3882       Advertising & Marketing     40.0   
1                   33.6  30547.9317    Logistics & Transportation     39.0   
2                   24.9  23880.4852               Food & Beverage    190.0   
7                   23.3  12700.6588            Telecommunications     72.0   
13                  19.1  10561.7147            Financial 

## Yes, There are 24 companies which was founded last year and yet manage to secure its postion in top 100 ranked companies.